news-explorer/src/utils/auth.js

```diff
@@ -0,0 +1,48 @@
+import mainApi from './mainApi';
+
+export const BASE_URL = 'https://api.new-explorer.students.nomoredomainssbs.ru';
+
+export function register(email, username, password) {
+	return mainApi
+		.customFetch(`${BASE_URL}/signup`, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({
+				email: email,
+				password: password,
+				name: username,
+			}),
+		})
+
+		.then((data) => {
+			if (data) {
+				return data;
+			} else {
+				throw new Error('User with this email address already exists');
+			}
+		});
+}
+
+export function authorize(email, password) {
+	return mainApi
+		.customFetch(`${BASE_URL}/signin`, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({email, password}),
+		})
+		.then((data) => {
+			if (data.token) {
+				localStorage.setItem('jwt', data.token);
+				return data.token;
+			} else {
+				throw new Error('the user with the specified email not found');
+			}
+		})
+		.catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` (auth) because you need to `catch` possible errors at the very end of any request and this end is in (`App.js`)  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.